tools/automation/sections/device_section.py

```diff
@@ -0,0 +1,212 @@
+"""
+  A collection of classes to represent the device section of the YAML.
+
+  They are responsible for filling up the section with data.
+
+  They also hold the logic for how to read the input files in order to
+  retrieve the correct data for a device.
+"""
+
+import re
+import abc
+
+from building_blocks.device import Device
+
+"""
+  Base class for the device section.
+"""
+class DeviceSection:
+
+  def __init__(self, files):
+    self._devices = []
+
+  # Logic for reading the excel data to fill up device data.
+  @abc.abstractmethod
+  def _create_device(self, data):
+    pass

```

Comment:
nit: maybe raise a NotImplementedError instead of passing silently

https://docs.python.org/3/library/exceptions.html#NotImplementedError